pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -12772,3 +12773,138 @@ func TestCreateTableLevelChangefeedWithDBPrivilege(t *testing.T) {
+				sqlDB.Exec(t, `INSERT INTO bar VALUES (0, 'initial')`)
+			}()
+			foo := feed(t, f, `CREATE CHANGEFEED FOR DATABASE d`)
+			defer closeFeed(t, foo)
+			assertPayloads(t, foo, []string{
+				`foo: [0]->{"after": {"a": 0, "b": "updated"}}`,
+			})
+			wg.Wait()
+		}
+		cdcTest(t, testFnAdd, withKnobsFn(knobsFn))
+	})
+
+	createAnotherTable = make(chan struct{}, 1)
+	// This tests adding tables to a database while a db-level changefeed is
+	// being resumed.
+	t.Run("add a table during resume", func(t *testing.T) {
+		testFnAddDuringResume := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+			sqlDB := sqlutils.MakeSQLRunner(s.DB)
+			sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING)`)
+
+			baseFeed := feed(t, f, `CREATE CHANGEFEED FOR DATABASE d`)
+			jobFeed := baseFeed.(cdctest.EnterpriseTestFeed)
+			defer closeFeed(t, baseFeed)
+			<-createAnotherTable
+
+			require.NoError(t, jobFeed.Pause())
+
+			sqlDB.Exec(t, `INSERT INTO foo VALUES (0, 'initial')`)
+			sqlDB.Exec(t, `CREATE TABLE bar (a INT PRIMARY KEY, b STRING)`)
+			sqlDB.Exec(t, `INSERT INTO bar VALUES (0, 'initial')`)
+			var wg sync.WaitGroup
+			wg.Add(1)
+			// goroutine to make tables. Signaled to begin creating tables from test
+			// knob callback after resumeWithRetries picks the first targets.
+			go func() {
+				defer wg.Done()
+				<-createAnotherTable
+				sqlDB.Exec(t, `CREATE TABLE baz (a INT PRIMARY KEY, b STRING)`)
+				sqlDB.Exec(t, `INSERT INTO baz VALUES (0, 'initial')`)
+			}()
+			sqlDB.Exec(t, `RESUME ALL CHANGEFEED JOBS`)
+			assertPayloads(t, baseFeed, []string{
+				`foo: [0]->{"after": {"a": 0, "b": "initial"}}`,
+				// Don't expect bar because it was created after the changefeed was
+				// paused. The changefeed resumes from where it left off.
+			})
+			wg.Wait()
+		}
+		cdcTest(t, testFnAddDuringResume, withKnobsFn(knobsFn), feedTestEnterpriseSinks)
+	})
+
+	// This tests the specific code path for fetching descriptors for enriched
+	// changefeeds.
+	t.Run("add a table during creation of enriched changefeed", func(t *testing.T) {
+		testFnAddEnriched := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+			sqlDB := sqlutils.MakeSQLRunner(s.DB)
+			sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING)`)
+			sqlDB.Exec(t, `INSERT INTO foo VALUES (0, 'initial')`)
+
+			var wg sync.WaitGroup
+			wg.Add(1)
+			// goroutine to make tables. Signaled to begin creating tables from test
+			// knob callback after resumeWithRetries picks the first targets.
+			go func() {
+				defer wg.Done()
+				<-createAnotherTable
+				sqlDB.Exec(t, `CREATE TABLE bar (a INT PRIMARY KEY, b STRING)`)
+				sqlDB.Exec(t, `INSERT INTO bar VALUES (0, 'initial')`)
+			}()
+			foo := feed(t, f, `CREATE CHANGEFEED FOR foo WITH envelope='enriched', enriched_properties='source'`)
+			defer closeFeed(t, foo)
+			assertPayloadsEnriched(t, foo, []string{
+				`foo: {"a": 0}->{"after": {"a": 0, "b": "initial"}, "op": "c"}`,
```

Comment:
Could you add a similar comment as you did in the previous subtest about not expecting bar to show up?